File: Same Tree/Same Tree.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param p, a tree node
    # @param q, a tree node
    # @return a boolean
    def isSameTree(self, p, q):
        if(p == None):
            if(q != None):
                return False
            else:
                return True
        elif(q == None):
            if(p != None):
                return False
            else:
                return True
        else:
            return (self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right) and (p.val == q.val))
            

File: Same Tree/Same Tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:

    bool recur(TreeNode* p, TreeNode* q){
        if((!p && q) || (p && !q))  return false;
        if(!p && !q)            return true;
        if(p->val != q->val)    return false;
        return (recur(p->left, q->left) && recur(p->right, q->right));
    }

    bool isSameTree(TreeNode *p, TreeNode *q) {
        recur(p, q);
    }
};
